test/help.test.js

```diff
@@ -51,36 +51,40 @@ describe('loopback generator help', function() {
 
   it('prints help message with lb if invoked from loopback-cli', function() {
     process.env.SLC_COMMAND = 'loopback-cli';
-    var gen = givenGenerator('app', ['--help']);
+    var gen = givenGenerator('app');
     var helpText = gen.help();
     debug('--HELP TEXT--\n', helpText);
+    // Jannyhou: to be fixed!
     assert(helpText.indexOf(' lb ') !== -1,
       '"lb" should be used');
     assert(helpText.indexOf('Available commands') !== -1,
       '"Available commands" should be used');
 
     assert(helpText.indexOf(' slc ') === -1,
       '"slc" should not be present');
+    // Jannyhou: to be fixed!
     assert(helpText.indexOf(' yo ') === -1,
       '"yo" should not be present');
     assert(helpText.indexOf('Available generators') === -1,
       '"Available generators" should not be present');
   });
 
+  // Jannyhou: to be fixed!
   describe('prints right help message for each generator', function() {
+    // var CMD_NAMES = ['acl'];
     var CMD_NAMES = ['acl', 'app', 'boot-script', 'datasource',
       'export-api-def', 'middleware', 'model', 'property', 'relation',
       'remote-method', 'swagger', 'bluemix'];
 
     CMD_NAMES.forEach(function(name) {
       it('prints right help message for generator ' + name, function() {
         process.env.SLC_COMMAND = 'slc';
-        var gen = givenGenerator(name, ['--help']);
+        var gen = givenGenerator(name);
         var helpText = gen.help();
         var helpFileName = 'loopback_' + name + '_help.txt';
         var helpFilePath = '../fixtures/help-texts/' + helpFileName;
         var output = fs.readFileSync(helpFilePath, 'utf8');
-        assert.equal(output, helpText);
+        assert.equal(helpText, output);
```

Comment:
The arguments are switched

Reply:
@shimks Yep, the old code was wrong. Read the error message you will see, the first argument is expected result, while the second one is the actual result.